specs/applicationdata/ApplicationData.md

```diff
@@ -0,0 +1,428 @@
+      [RoamingStorageQuota](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.roamingstoragequota),
+      [Version](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.version),
+      [SetVersionAsync()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.setversionasync),
+      [SignalDataChanged()](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.signaldatachanged)
+      and
+      [DataChanged](https://learn.microsoft.com/uwp/api/windows.storage.applicationdata.datachanged). See [5.1. Platform Support](#51-platform-support) for more details.
+* **GetDefault() works for all packaged apps** -- Equivalent to .Current but works for all packaged
+  apps (not just callers running in an AppContainer). See
+  [3.3. Reliable access](#33-reliable-access).
+* **`GetForPackageFamily()`** -- Equivalent to Windows.Storage.ApplicationDataManager.CreateForPackageFamily().
+* **`GetForUser()`** -- Synchronous equivalent to GetForUserAsync().
+* **`GetForUnpackaged()`** -- Equivalent functionality for unpackaged apps. See
+  [3.5. Unpackaged app data stores](#35-unpackaged-app-data-stores).
+
+## 3.2. *Path properties
+
+New *Path equivalents to Windows.Storage.ApplicationData's *Folder but returning a `String` rather than `StorageFolder`.
+
+## 3.3. Reliable access
+
+`Windows.Storage.ApplicationData.Current` only works for packaged apps **running in an AppContainer**. Processes not running in an AppContainer (e.g. MediumIL) must use the equivalent `Windows.Management.Core.ApplicationDataManager.CreateForPackageFamily(Windows.ApplicationModel.Package.Current.Id.FamilyName)`.
+
+Microsoft.Windows.Storage.ApplicationData provides consistent behavior as developers expect e.g. `ApplicationData.Current` works regardless if the caller's running in an AppContainer or not.
+
+## 3.4. Per-Machine data store
+
+Package families can optionally have a per-machine data store.
+
+### 3.4.1. Machine Path/Folder
+
+The path for a package family’s "Machine Folder" is located at `%ProgramData%\Microsoft\Windows\AppRepository\ApplicationData\...packagefamilyname...\Machine`.
+
+This directory is ACL’d similarly to a package’s System Metadata directory (`%ProgramData%\Microsoft\Windows\AppRepository\Packages\...pkgfullname...`) e.g.
+
+```
+BUILTIN\Users:(OI)(CI)(Rc,S,RD,REA,X,RA)
+S-1-15-3-2977037414-864741429-1129033548-1928484290-1803339615-1058153653-556172075:(OI)(CI)(RX)
+BUILTIN\Users:(OI)(CI)(R)
+NT SERVICE\TrustedInstaller:(I)(F)
+NT SERVICE\TrustedInstaller:(I)(CI)(IO)(F)
+NT AUTHORITY\SYSTEM:(I)(F)
+NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
+BUILTIN\Administrators:(I)(RX)
+BUILTIN\Administrators:(I)(OI)(CI)(IO)(GR,GE)
+S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204:(I)(RX)
+S-1-15-3-1024-3635283841-2530182609-996808640-1887759898-3848208603-3313616867-983405619-2501854204:(I)(OI)(CI)(IO)(GR,GE)
+```
```

Comment:
What's a use case for this? My naive read is that only TrustedInstaller and System get write access. How is an app supposed to use this?

Reply:
Yes, but Admins get read/execute access and Users get read access.

An app can write content if it runs elevated, or is a Service running as LocalSystem, etc. The latter is the driving use case -- a packaged Service writing content in its package family's MachinePath for read+execute access by other processes (not necessarily in its package or even packaged at all).

And in the cases where some of the content needs access other than the default rights we create, the Service will create a subdir with the needed permissions. That's no different than some packaged apps do today -- create ApplicationData.LocalFolder\somedir ACL'd with different (non-default) permissions for their scenario.

Reply:
Consider updating the text so the "why should you care" is more clear. Otherwise apps will think "oh hey I can easily break the user security model and share stuff with other users on the same machine!"

Reply:
CONSIDER: Add more text about WHY and consider-security-implications